src/D2L.Security.OAuth2/Keys/Caching/InMemoryPublicKeyCache.net4x.cs

```diff
@@ -26,5 +27,13 @@ internal sealed partial class InMemoryPublicKeyCache : IInMemoryPublicKeyCache {
 			var result = m_cache.Get( BuildCacheKey( srcNamespace, keyId ) ) as D2LSecurityToken;
 			return result;
 		}
+
+		private static void EvictionCallback( CacheEntryRemovedArguments args ) {
+			if( args.CacheItem.Value is not D2LSecurityToken key ) {
+				return;
+			}
+
+			key.Dispose();
+		}

```

Comment:
Can check EvictionReason here and in the net6 path to not dispose on explicit removals or something, but I figure if we do the `Ref()` within the `Set` methods here, then the cache owns the ref and should do the `Dispose()` whenever it leaves.